test/jdk/javax/swing/border/LineBorder/ScaledLineBorderTest.java

```diff
@@ -74,51 +84,52 @@ public static void main(String[] args) throws Exception {
     }
 
     private static void testScaling(boolean showFrame, boolean saveImages) {
-        JComponent content = createUI();
-        if (showFrame) {
-            showFrame(content);
+        for (int thickness : thickness) {
+            TestUI testUI = createUI(thickness);
+            if (showFrame) {
+                showFrame(testUI.content);
+            }
+
+            List<TestImage> images = paintToImages(testUI, saveImages);
+            verifyBorderRendering(images, saveImages);
+        }
+
+        if (errorCount > 0) {
+            throw new Error("Test failed: "
+                    + errorCount + " error(s) detected - "
+                    + errorMessage);
         }
 
-        paintToImages(content, saveImages);
-        verifyBorderRendering(saveImages);
     }
 
-    private static void verifyBorderRendering(final boolean saveImages) {
-        String errorMessage = null;
-        int errorCount = 0;
-        for (int i = 0; i < images.size(); i++) {
-            BufferedImage img = images.get(i);
-            double scaling = scales[i];
-            try {
-                int thickness = (int) Math.floor(scaling);
+    private static String errorMessage = null;
+    private static int errorCount = 0;
 
-                checkVerticalBorders(SIZE.width / 2, thickness, img);
+    private static void verifyBorderRendering(final List<TestImage> images,
+                                              final boolean saveImages) {
+        for (TestImage test : images) {
+            final BufferedImage img = test.image;
+            final int effectiveThickness = clipRound(test.thickness * test.scale);
+            try {
+                checkVerticalBorders((int) (SIZE.width * test.scale / 2), effectiveThickness, img);
 
-                for (Point p : panelLocations) {
-                    int y = (int) (p.y * scaling) + SIZE.height / 2;
-                    checkHorizontalBorder(y, thickness, img);
+                for (Point p : test.panelLocations) {
+                    int y = (int) (p.y * test.scale) + (int) (SIZE.height * test.scale / 2);
```

Comment:
This expression can be simplified:
```suggestion
                    int y = (int) ((p.y + (SIZE.height / 2)) * test.scale);
```
This way may look clearer: find the vertical centre of the panel, taking into account the scale applied.